src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -147,12 +147,39 @@ public void InitializeService(ServiceHost serviceHost)
                     DataTable dt = tuple.Item2;
                     int count = dt.Rows.Count;
                     var agentJobs = new List<AgentJobHistoryInfo>();
+
+                    var agentStepMap = new Dictionary<DateTime, List<AgentJobStep>>();
```

Comment:
is RunDate the best way to group these?  I assume these nothing like an ExecutonId field or something?

Reply:
RunDate was the only thing I could find to tie the steps to the jobs. They all have the same `jobID`'s, and no other `ID` field.